### PR TITLE
[COMPILE] workflow for deepspeed + torch.compile

### PR DESCRIPTION
**Pull request by YizhouZ**
We use simple model + deepspeed zero 3 + torch.compile and count graph break numbers to demonstrate current status of combing deepspeed + torch.compile.

### CODE REVIEW

File: .github/workflows/xpu-compile.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  issues: write
+
+jobs:
+  compile-tests:
+    runs-on: [self-hosted, intel, xpu]
```
Comment (loadams) [inline .github/workflows/xpu-compile.yml:21]: @YizhouZ - does it make more sense to add this to its own workflow, or to add it to the existing xpu workflow?

File: tests/torch_compile/test_compile.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    # runs backpropagation
+    model_engine.backward(loss)
+    # weight update
+    model_engine.step()
+
+dynamo_stats = get_dynamo_stats()
+dynamo_stats.subtract(start_stats)
+
+if comm.get_rank() == 0:
+    print(dynamo_stats)
```
Comment (loadams) [inline tests/torch_compile/test_compile.py:103]: @YizhouZ - could you run the pre-commit formatter on the PR, that will resolve the formatting/python issues.

**Comment by delock** [comment]
I also have a question for discussion.  Initially I thought of reference test result of this workflow (as number of graph breaks) on a page, a little bit like workflow badges on DeepSpeed README page.  But I didn't find proper tool allowing us to do that.  Is there a pointer or direction where we should look at?  Thanks!

File: tests/torch_compile/test_compile.py (hunk 2 above)
Comment (YizhouZ) [reply tests/torch_compile/test_compile.py:103]: > @YizhouZ - could you run the pre-commit formatter on the PR, that will resolve the formatting/python issues.

sure!

File: .github/workflows/xpu-compile.yml (hunk 1 above)
Comment (YizhouZ) [reply .github/workflows/xpu-compile.yml:21]: > @YizhouZ - does it make more sense to add this to its own workflow, or to add it to the existing xpu workflow?

you mean adding a new worker for this workflow?

**Comment by delock** [comment]
xpu_compile workflow now passed.  Now the graph break number shows in test summary
https://github.com/microsoft/DeepSpeed/actions/runs/11044896580?pr=6570

**Review by tohtana — APPROVED** [review]
Thank you @YizhouZ, looks good to me. It is great to have utilities like this in our workflows.

**Pull request merged**